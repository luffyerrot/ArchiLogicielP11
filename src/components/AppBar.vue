<template>
    <v-app-bar>
        <template v-slot:prepend>
            <v-app-bar-nav-icon
                variant="text"
                @click.stop="drawer = !drawer"
            />
        </template>
        <template v-slot:append>
            <v-btn
                :prepend-icon="props.theme === 'light' ? 'mdi-weather-sunny' : 'mdi-weather-night'"
                text="Changer le thÃ¨me"
                slim
                @click="$emit('changeTheme')"
            />
            <p class="mx-4">{{ user.mail }}</p>
        </template>
        <v-app-bar-title
            class="clickable"
            text="Service MedHead"
            @click="goTo('/')"
        />
    </v-app-bar>
    <v-navigation-drawer
        v-model="drawer"
        temporary
    >
        <v-list
            v-for="(item, index) in items"
            :key="'menu' + index"
        >
            <v-list-item
                :title="item.title"
                @click="goTo(item.destination)"
            />
        </v-list>
    </v-navigation-drawer>
</template>

<script setup lang="ts">
    import router from '@/router';
    import { useUserStore } from '@/stores/user';

    defineEmits(['changeTheme'])

    const props = defineProps<{
        theme: string
    }>()

    const user = useUserStore();

    const drawer = ref(false);
    const items = computed(() => {
        const baseItems = [
            {
                title: 'Accueil',
                destination: '/'
            },
            {
                title: 'Recherche',
                destination: '/mapfinder'
            }
        ];
        const loggedInItems = [
            {
                title: 'S\'enregistrer',
                destination: '/register'
            },
            {
            title: 'Connexion',
            destination: '/login'
            }
        ];
        const loggedOutItems = [
            {
                title: 'Se deconnecter',
                destination: '/logout'
            }
        ];
        return [...(user.isLogged() ? loggedOutItems : loggedInItems), ...baseItems]
    });

    function goTo(destination: string) {
        router.push(destination);
    }
</script>

<style scoped lang="scss">
    .clickable {
        cursor: pointer;
    }
</style>