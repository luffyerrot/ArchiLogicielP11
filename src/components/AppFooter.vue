<template>
  <v-footer height="60" app elevation="6">
    <p class="message">Un super footer</p>
  </v-footer>
</template>

<script setup lang="ts">
</script>

<style scoped lang="scss">
    .message {
        text-align: center;
    }
</style>