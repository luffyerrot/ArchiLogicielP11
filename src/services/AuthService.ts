import { ConnexionDto, InformationConnexion } from "@/types";
import axios, { AxiosPromise } from "axios";

export function subscribe(connexion: ConnexionDto): AxiosPromise<ConnexionDto> {
	return axios.post('/user/auth/subscribe', connexion);
}

export function login(connexion: ConnexionDto): AxiosPromise<InformationConnexion> {
	return axios.post('/user/auth/login', connexion);
}