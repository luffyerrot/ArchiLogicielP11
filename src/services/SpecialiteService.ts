import { GoogleMapPath, GroupeSpecialiteDto, HopitalDto, MapPathDto, SpecialiteDto } from "@/types";
import axios, { AxiosPromise } from "axios";

export function getAllGroupe(): AxiosPromise<GroupeSpecialiteDto[]> {
	return axios.get<GroupeSpecialiteDto[]>('/api/groupes');
}

export function getSpecialitesByGroupeId(groupeId: number): AxiosPromise<SpecialiteDto[]> {
	return axios.get<SpecialiteDto[]>('/api/specialites?id=' + groupeId);
}

export function getHopitauxBySpecialiteId(specialiteId: number): AxiosPromise<HopitalDto[]> {
	return axios.get<HopitalDto[]>('/api/hopitaux?id=' + specialiteId);
}

export function getPath(mapPath: MapPathDto): AxiosPromise<GoogleMapPath> {
	return axios.put<GoogleMapPath>('/api/path', mapPath);
}