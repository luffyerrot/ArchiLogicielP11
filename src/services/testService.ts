import axios, { AxiosPromise } from "axios";

export const getTest = (): AxiosPromise<string> => {
	return axios.get('test/');
}