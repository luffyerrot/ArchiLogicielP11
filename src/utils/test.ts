import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}