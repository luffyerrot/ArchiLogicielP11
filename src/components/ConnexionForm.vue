<template>
    <h1 class="message"> {{ props.register ? 'Enregistrement' : 'Connexion' }} </h1>
    <v-card class="mx-auto px-6 py-8 mt-4" max-width="344">
      <v-form
        v-model="form"
        @submit.prevent="props.register ? onRegister() : onLogin()"
      >
        <v-text-field
          v-model="mail"
          :readonly="loading"
          :rules="[required]"
          class="mb-2"
          label="E-mail"
          type="email"
          clearable
        />

        <v-text-field
          v-model="password"
          :readonly="loading"
          :rules="[required]"
          label="Mot de passe"
          type="password"
          placeholder="Entrer le mot de passe"
          clearable
        />

        <br>

        <v-btn
          :disabled="!form"
          :loading="loading"
          color="success"
          size="large"
          type="submit"
          variant="elevated"
          block
        >
          {{ props.register ? "S'enregistrer" : "Se connecter" }}
        </v-btn>
      </v-form>
      <div v-if="msgConnexionError !== null">
        <p
          class="message mt-4"
          style="color: red"
        >
          {{ msgConnexionError }}
        </p>
      </div>
    </v-card>
</template>
  
<script setup lang="ts">
    import router from '@/router';
import { login, subscribe } from '@/services/AuthService';
    import { useUserStore } from '@/stores/user';

    const props = defineProps<{
        register: boolean
    }>()

    const user = useUserStore()

    const msgConnexionError: Ref<string | null> = ref(null);
    const form: Ref<boolean> = ref(false);
    const mail: Ref<string | null> = ref(null);
    const password: Ref<string | null> = ref(null);
    const loading: Ref<boolean> = ref(false);

    const connexionDto = computed(() => ({ mail: mail.value ? mail.value : '', motdepasse: password.value ? password.value : '' }));

    function onRegister() {
        loading.value = true;
        subscribe(connexionDto.value)
        .then(() => {
            msgConnexionError.value = null;
            router.push('/login');
        })
        .catch(() => msgConnexionError.value = 'Erreur lors de l\'inscription')
        .finally(() => loading.value = false);
    };
    
    function onLogin() {
        loading.value = true;
        login(connexionDto.value)
        .then(response => {
            user.updateUser(response.data);
            router.push('/mapfinder');
        })
        .catch(() => msgConnexionError.value = 'Erreur lors de la connexion')
        .finally(() => loading.value = false);
    };

    const required = (value: string) => {
        return !!value || 'Le champ est requis'
    };
</script>

<style scoped lang="scss">
  .message {
    text-align: center;
  }
</style>