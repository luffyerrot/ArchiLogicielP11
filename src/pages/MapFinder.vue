<template>
    <v-card class="mx-auto px-6 py-8 mt-4" max-width="800">
        <GoogleMap
            :api-key="apiKey"
            style="width: 100%; height: 500px"
            :center="mapCenter"
            :zoom="10"
        >  
            <Polyline :options="drivingPath" />
        </GoogleMap>
        <v-btn @click="findPath">Chercher</v-btn>
    </v-card>
    <div v-if="msgFindError !== null">
        <p
            class="message ma-4"
            style="color: red"
        >
            {{ msgFindError }}
        </p>
    </div>
</template>

<script setup lang="ts">
    import { getPath } from '@/services/SpecialiteService';
    import { PointDto } from '@/types';
    import { GoogleMap, Polyline } from 'vue3-google-map'
    
    const msgFindError: Ref<string | null> = ref(null);
    const mapCenter = { lat: 47.36888117749755, lng: -0.49352569804404806 };
    const apiKey = 'AIzaSyB86d0Y3Drd6HxhN2aYDI-hnkLf6rw1WmE';
    const origin: PointDto = { latitude: 47.35866413146318, longitude: -0.5505390654434574 }
    const route = useRoute();
    const targetLatitude: Ref<number | undefined> = ref(+(route.query.latitude as string));
    const targetLongitude: Ref<number | undefined> = ref(+(route.query.longitude as string));

    const drivingCoordinates: Ref<any[]> = ref([]);

    const drivingPath = computed(() => {
        return {
            path: drivingCoordinates.value,
            geodesic: true,
            strokeColor: '#FF0000',
            strokeOpacity: 1.0,
            strokeWeight: 2
        }
    });

    const findPath = () => {
        if (targetLatitude.value && targetLongitude.value) {
            getPath({ origin, target: { latitude: targetLatitude.value, longitude: targetLongitude.value } as any as PointDto })
                .then(response => {
                    msgFindError.value = null;
                    const leg = response.data.routes[0].legs[0];
                    drivingCoordinates.value = [ leg.start_location, ...leg.steps.map(step => step.end_location) ];
                })
                .catch(() => msgFindError.value = 'Erreur lors de la récupération des infos sur le chemin');
        } else {
            msgFindError.value = 'Impossible d\'effectuer la recherche';
        }
    }
</script>

<style scoped lang="scss">
    .message {
        text-align: center;
    }
</style>