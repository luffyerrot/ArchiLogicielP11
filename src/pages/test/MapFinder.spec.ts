import { factory, renderCorrectly } from "@/utils/test";
import { describe, expect, it, vi } from "vitest";
import MapFinder from "../MapFinder.vue";
import { VBtn } from "vuetify/components";
import { getPath } from "@/services/SpecialiteService";

describe('Page de connexion', () => {
    vi.mock('vue-router', () => ({
        useRoute: vi.fn(() => ({
            query: {
                latitude: 0.1,
                longitude: 0.1
            }
        }))
    }))
    const wrapper = factory(MapFinder);

    renderCorrectly(wrapper);

    it('Appel le service de recherche', async () => {
        vi.mock('@/services/SpecialiteService', {
            spy: true
        })

        const btn = wrapper.findComponent(VBtn);

        expect(btn.exists()).toBeTruthy();

        await btn.trigger('click');

        expect(getPath).toHaveBeenCalledOnce();
    })
})