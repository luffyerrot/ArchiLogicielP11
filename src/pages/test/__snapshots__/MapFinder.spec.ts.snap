// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page de connexion > Bon affichage de la page 1`] = `
"<div data-v-fb44f027="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto px-6 py-8 mt-4" style="max-width: 800px;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-d05fc6bc="" data-v-fb44f027="" style="width: 100%; height: 500px;">
    <div data-v-d05fc6bc="" class="mapdiv"></div>
    <!---->
  </div><button data-v-fb44f027="" type="button" class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
    <!----><span class="v-btn__content" data-no-activator="">Chercher</span>
    <!---->
    <!---->
  </button>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>
<!--v-if-->"
`;
